flink-libraries/flink-table/src/main/scala/org/apache/flink/table/plan/schema/StreamTableSourceTable.scala

```diff
@@ -39,6 +39,8 @@ class StreamTableSourceTable[T](
     val fieldCnt = fieldNames.length
 
     val rowtime = tableSource match {
+      case emptyStringTimeSource : DefinedRowtimeAttribute if emptyStringTimeSource.getRowtimeAttribute.trim.equals("")  =>
```

Comment:
I checked the null timeAttribute first, and then check empty string timeAttribute when it is not null, if it is empty string, throw TableException.

Please check again.